src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -282,12 +284,23 @@ private AdjustmentRule GetAdjustmentRuleForAmbiguousOffsets(DateTime adjustedTim
         /// If the specified rule is the first AdjustmentRule, or it isn't in _adjustmentRules,
         /// then the specified rule is returned.
         /// </summary>
-        private AdjustmentRule GetPreviousAdjustmentRule(AdjustmentRule rule)
+        private AdjustmentRule GetPreviousAdjustmentRule(AdjustmentRule rule, int? ruleIndex)
         {
+            Debug.Assert(rule.NoDaylightTransitions, "GetPreviousAdjustmentRule should only be used with NoDaylightTransitions rules.");
+
+            if (ruleIndex.HasValue && 0 < ruleIndex.Value && ruleIndex.Value < _adjustmentRules.Length)
+            {
+                return _adjustmentRules[ruleIndex.Value - 1];
```

Comment:
I'm not familiar with what the rule indices mean, but is it correct that we compare `ruleIndex < _adjustmentRules.Length` but then access `_adjustmentRules[ruleIndex.Value - 1]`?  Naively in such situations I'd have expected it to be `<=` instead of `<`, or to be `ruleIndex.Value` rather than `ruleINdex.Value-1`.